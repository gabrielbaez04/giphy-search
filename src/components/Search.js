import React from 'react';
import PropTypes from 'prop-types';

class Search extends React.Component {
    state = { term: '' }

    onInputChange = (event) => {
        const { value, name } = event.target;
        this.setState({ [name]: value });
        value.length > 2 && this.props.onTermChange(value);
    }

    render() {
        return (
            <div className="search">
                <input name="term" placeholder="Enter text to search for gifs!" onChange={this.onInputChange} />
            </div>
        );
    }
}

Search.propTypes = {
    onTermChange: PropTypes.func.isRequired
}

export default Search;