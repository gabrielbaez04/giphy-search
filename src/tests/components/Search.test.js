import { shallow } from 'enzyme';
import Search from '../../components/Search';
import React from 'react';

test('should render Search correctly', () => {
    const wrapper = shallow(<Search onTermChange={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
});

test('should set term on input change', () => {
    const value = 'New term';
    const wrapper = shallow(<Search onTermChange={jest.fn()} />);
    wrapper.find('input').at(0).simulate('change', {
        target: { value, name: 'term' },

    });
    expect(wrapper.state('term')).toBe(value);
})
