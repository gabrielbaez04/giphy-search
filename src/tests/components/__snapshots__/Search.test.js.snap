// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Search correctly 1`] = `
<div
  className="search"
>
  <input
    name="term"
    onChange={[Function]}
    placeholder="Enter text to search for gifs!"
  />
</div>
`;
